Source/missiles.cpp

```diff
@@ -4773,48 +4773,51 @@ void MI_Rportal(int i)
 	PutMissile(i);
 }
 
-void ProcessMissiles()
+static void DeleteMissiles()
 {
-	for (int i = 0; i < ActiveMissileCount; i++) {
-		dFlags[Missiles[ActiveMissiles[i]].position.tile.x][Missiles[ActiveMissiles[i]].position.tile.y] &= ~BFLAG_MISSILE;
-		dMissile[Missiles[ActiveMissiles[i]].position.tile.x][Missiles[ActiveMissiles[i]].position.tile.y] = 0;
-		if (Missiles[ActiveMissiles[i]].position.tile.x < 0 || Missiles[ActiveMissiles[i]].position.tile.x >= MAXDUNX - 1 || Missiles[ActiveMissiles[i]].position.tile.y < 0 || Missiles[ActiveMissiles[i]].position.tile.y >= MAXDUNY - 1)
-			Missiles[ActiveMissiles[i]]._miDelFlag = true;
-	}
-
 	for (int i = 0; i < ActiveMissileCount;) {
 		if (Missiles[ActiveMissiles[i]]._miDelFlag) {
 			DeleteMissile(ActiveMissiles[i], i);
 		} else {
 			i++;
 		}
 	}
+}
+
+void ProcessMissiles()
+{
+	for (int i = 0; i < ActiveMissileCount; i++) {
+		auto &missile = Missiles[ActiveMissiles[i]];
+		const auto &position = missile.position.tile;
+		dFlags[position.x][position.y] &= ~BFLAG_MISSILE;
+		dMissile[position.x][position.y] = 0;
+		if (!InDungeonBounds(position))
+			missile._miDelFlag = true;

```

Comment:
Again, not something you've introduced in this PR but it stands out now that you've cleaned up the code. This is a very strange check, position is assumed to be valid and used to index into the dFlags and dMissile arrays and then checked after it would cause out of bounds writes? From reading the code the only time the !InDungeonBounds check accomplishes anything is if the missile moves into row/col 0.

We should be able to remove the InDungeonBounds check here once the missile procedures handle bounds checking properly. I can log an issue to track that once I've had a bit of time to think about the wording 😄

Reply:
Yeah I only wanted to make this code more readable since it smelled a bit :D